add agent instructions and shared contribution guide for MaMMoS (#218)
MaMMoS-project/mammos-entity: AGENTS.md

# Agent Instructions for mammos-entity

> This file is only for AI coding agents. Human-facing package guidance belongs in

Reviewer: Maybe (as it is for both humans and AI)

```suggestion
This file is only for AI coding agents. General package guidance belongs in
```

`CONTRIBUTING.md`; shared MaMMoS standards belong in
`CONTRIBUTING-MaMMoS.md`.

> Read `CONTRIBUTING.md` and `CONTRIBUTING-MaMMoS.md` before editing.

Reviewer: Maybe also the (yet to be written) section `README.md#About`?

> Read `CONTRIBUTING.md` and `CONTRIBUTING-MaMMoS.md` before editing.

Author: ```suggestion
Read `CONTRIBUTING.md`, `CONTRIBUTING-MaMMoS.md`, and `README.md` before editing.
```

This repository must work as a standalone checkout. Do not assume
`mammos-devtools` or sibling repositories are present. Use this repository's
`pyproject.toml`, pixi tasks, tests, examples, and docs as the local source of

> truth.

Reviewer: ```suggestion
truth unless explicitly instructed otherwise.
```

Sometimes it is necessary to use the pixi environment from devtools to test changes that affect multiple packages simultaneously.

If this checkout is located at `mammos-devtools/packages/mammos-entity`, also

> read `../../AGENTS.md` for umbrella-repository guidance.

Reviewer: What would go into that AGENTS.md file?

Edit: I took a look at the other PR. Is it required to also point to that base AGENTS.md file?
